pkgs/top-level/all-packages.nix

```diff
@@ -1610,7 +1610,9 @@ with pkgs;
 
   ddrescue = callPackage ../tools/system/ddrescue { };
 
-  deluge = python2Packages.deluge; # Package should be moved out of python-packages.nix
+  deluge = callPackage ../applications/networking/p2p/deluge {
+    pythonPackages = python2Packages;
```

Comment:
The explicit version is used when it is also needed. If the package may work with both versions then  no version is used. Its kind of analogous to PEP 394 https://www.python.org/dev/peps/pep-0394/.